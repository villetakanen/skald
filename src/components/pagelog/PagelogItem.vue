<template>
  <div class="pagelog-item">
    <div class="pagelog-type">
      <v-icon v-if="item.action === 'create'">mdi-plus</v-icon>
      <v-icon v-if="item.action === 'update'">mdi-file-document-edit-outline</v-icon>
      <v-icon v-if="item.action === 'delete'">mdi-close</v-icon>
    </div>
    <div class="pagelog-details">
      <p class="link"><a :href="'/#/v/' + item.siteid + '/' + item.pageid">{{ item.siteid + '/' + item.pageid }}</a><br/>
      {{ toDate(item.timestamp.seconds) }} {{item.creatorNick}}</p>
    </div>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import VueCompositionApi, { defineComponent, ref } from '@vue/composition-api'
Vue.use(VueCompositionApi)

export default defineComponent({
  props: {
    item: {
      type: [Object],
      required: true
    }
  },
  setup (props) {
    const toDate = (seconds:number|null|undefined) => {
      if (seconds === null || typeof seconds === 'undefined') return '...'
      var lastChangeDate = new Date(1970, 0, 1) // Epoch
      lastChangeDate.setSeconds(seconds)
      const s = lastChangeDate.toISOString().split('T')
      return s[0] + ' ' + s[1].substring(0, 8)
    }
    return { toDate }
  }
})
</script>

<style scoped>
div.pagelog-item{
  display: flex;
  flex-direction: row;
}
div.pagelog-type{
  width: 50px;
  padding: 8px;
}
div.pagelog-entry p{
  margin: 0;
  padding: 0;
}
div.pagelog-entry p:last-of-type{
  padding-bottom: 8px;
}
</style>
