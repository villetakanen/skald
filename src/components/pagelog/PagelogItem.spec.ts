import { shallowMount } from '@vue/test-utils'
import PagelogItem from './PagelogItem.vue'

describe('Mounted App', () => {
  it('renders correctly', () => {
    const wrapper = shallowMount(PagelogItem, {
      mocks: {
        $t: () => {}
      },
      propsData: {
        item: {
          action: 'create',
          timestamp: {
            seconds: 1
          }
        }
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
