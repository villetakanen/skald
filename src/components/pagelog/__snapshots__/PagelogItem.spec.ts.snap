// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mounted App renders correctly 1`] = `
<div
  class="pagelog-item"
>
  <div
    class="pagelog-type"
  >
    <v-icon-stub>
      mdi-plus
    </v-icon-stub>
     
    <!---->
     
    <!---->
  </div>
   
  <div
    class="pagelog-details"
  >
    <p
      class="link"
    >
      <a
        href="/#/v/undefined/undefined"
      >
        undefined/undefined
      </a>
      <br />
      
    1969-12-31 22:00:01 
    </p>
  </div>
</div>
`;
